### ISSUE TITLE
Significant differences in reproduction results

### ISSUE THREAD

**Issue by xinyanghuang7** [Bostoncake/C-VisDiT#1]
Hello, author. Thank you very much for making your code and configuration files public. 

However, I encountered some issues when trying to reproduce your results. 

I changed the num_epochs in **config/office/A-D-1.json to 500 without making any other alterations**, but the final results I obtained differed significantly from the ones in your logs (**61.0 vs 74.1**). 

This has been troubling me for some time, and I hope to get your assistance.

At the same time, I have noticed slight differences between the current log file and the log you provided. 

I am looking forward to your answer!

![image](https://github.com/Bostoncake/C-VisDiT/assets/75032525/8e83841a-0ce5-4cd5-8582-6cabb34c8895)

**Comment by Bostoncake** [comment]
Hi Xinyang,

Thank you for paying attention to our work, and especially for pointing out the mistake in `config/office/A-D-1.json` (`num_epochs` should be 500 according to [PCS])! In fact, this codebase is mostly based on the codebase of [PCS], as we already stated in the `README.md`. At the time when we were getting familiar with reproducing the results in [PCS] (you may notice that **[PCS] is the only open-source FUDA solution before us**, since [CDS] did not publish their code at the time we worked for our solution), we noticed that due to the small size of the Office-31 dataset, the performance results could **fluctuate between several runs even if we use the same seed for initialization**. To ensure the fairness of comparison and get rid of the impact caused by randomness, we managed to find the corresponding seed for each adaptation **before we conduct our C-VisDiT**, where by "corresponding seed" we mean the seed that could **consistently reproduce the results of [PCS] and do not cause significant fluctuation** between different runs (and that is why you may see the ugly seed specifications in our configurations). 

We reproduce the results for [PCS] on CUDA Version: 11.4 and RTX 3090 GPUs. You may need to find the appropriate configuration in your own environment. 

Once again, our C-VisDiT is not affected by the randomness and could consistently produce performance gains. After setting up a consistent environment, or using the generalization pre-trained weights as in BrAD, as we stated in our paper, C-VisDiT shows significant, and consistent performance gains and greatly improves feature learning. We are planning to upload our scripts for BrAD and VisDA-C soon.  

Feel free to raise more questions if you have encountered any more problems.

**Comment by xinyanghuang7** [comment]
Thank you for your very quick answer!
Based on your answer, can I think that if I need to reproduce the corresponding results, I may need to run it multiple times under a fixed random seed or run multiple times under different random seeds.
An additional question, regarding the results of OFFICE-31, I am curious about the specific method you used to determine the random seed (e.g. 1607667202)?
Finally, I'm looking forward to your scripts for BrAD and VisDA-C :)

**Comment by Bostoncake** [comment]
You may need to reproduce PCS results first.
That is, if you find that our results are hard to reproduce, then most likely you cannot reproduce the results in the corresponding setting for PCS. What you need to do is to set our C-VisDiT loss weights zero in the configuration, run multiple experiments under different seeds until you can consistently reproduce the result for the setting in PCS, then add on our C-VisDiT modules. 

I know this might seem absurd, but this is currently the best solutions I could provide. We are still working on re-building the codebase to make it more convenient to reproduce both results in PCS and C-VisDiT. Please keep track of our latest update in this repo if you are interested!

**Comment by xinyanghuang7** [comment]
Thank you for your timely reply. I will always pay attention to your work!

Looking forward to your announcement of future work!

Good luck to you!
